### DISCUSSION TITLE
支付宝小程序上怎么实现手指左/右滑动 等 事件呢

### DISCUSSION THREAD

**Discussion by ChenYuZhu505812971** [galacean/discussion#11] [Mini Program（小程序）] [answered]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by cptbtptpbcptdtptp** [comment]
https://github.com/oasis-engine/engine/discussions/983

**Comment by cptbtptpbcptdtptp** [comment, marked as answer]
1. 首先，在小程序适配触控事件需要参考[小程序导出文档](https://galacean.antgroup.com/engine/docs/latest/cn/assets-build#%E5%B0%8F%E7%A8%8B%E5%BA%8F%E9%A1%B9%E7%9B%AE%E4%BD%BF%E7%94%A8-orbitcontrol)适配。
2. 在第一步的基础上，通过 [pointer](https://galacean.antgroup.com/engine/docs/latest/cn/input-pointer) 对象的 `deltaPosition` 来监听本帧或者这几帧的移动。
3. 根据业务的诉求，结合几帧触控的位移，判断是否左滑右滑。

可参考示例代码：
```typescript
class TouchControl extends Script {
     onUpdate() {
          const pointers = engine.inputManager.pointers;
          if(pointers && pointers.length > 0) {
                // 触控点在本帧的位置移动  pointers[0].deltaPosition
         }
     }
}
```
